File: GOL-C++/matrix/WilczeJagody.cpp
#include <iostream>

#include "Swiat.h"
#include "Organizm.h"
#include "Roslina.h"
#include "WilczeJagody.h"

WilczeJagody::WilczeJagody() : Roslina() {
	sila = 99;
}

WilczeJagody::WilczeJagody(int x, int y, Swiat* swiat) : Roslina(99, x, y, swiat) {}

char WilczeJagody::GetAwatar() const {
	return awatar;
}

string WilczeJagody::GetGatunek() const {
	return "Wilcze Jagody";
}

void WilczeJagody::akcja() {
	int ran = (rand() % 100) + 1;
	if (ran < ratio)
		Roslina::akcja();
}

void WilczeJagody::kolizja(Organizm* atakujacy) {
	swiat->zniszczOrganizm(atakujacy);
}

File: GOL-Java/WilczeJagody.java
package pl.edu.pg.eti.ksg.po.project2.rosliny;

import pl.edu.pg.eti.ksg.po.project2.Roslina;
import pl.edu.pg.eti.ksg.po.project2.Organizm;
import pl.edu.pg.eti.ksg.po.project2.Komentator;
import pl.edu.pg.eti.ksg.po.project2.Swiat;
import pl.edu.pg.eti.ksg.po.project2.Punkt;

import java.util.Random;
import java.awt.*;

public class WilczeJagody extends Roslina {
    private static final int sila = 99;
    private static final int inicjatywa = 0;

    public WilczeJagody(Swiat swiat, Punkt pozycja, int turaUrodzenia) {
        super(TypOrganizmu.WILCZE_JAGODY, swiat, pozycja, turaUrodzenia, sila, inicjatywa);
        setKolor(new Color(70, 29, 119));
        setSzansaRozmnazania(0.05);
    }


    @Override
    public void Akcja() {
        Random rand = new Random();
        int upperbound = 100;
        int tmpLosowanie = rand.nextInt(upperbound);
        if (tmpLosowanie < getSzansaRozmnazania() * 100) Rozprzestrzenianie();
    }

    @Override
    public String GetGatunek() {
        return "Wilcze jagody";
    }

    @Override
    public boolean SpecjalneDzialaniePodczasAtaku(Organizm atakujacy, Organizm ofiara) {
        Komentator.DodajKomentarz(atakujacy.OrganizmToString() + " zjada " + this.OrganizmToString());
        if (atakujacy.getSila() >= 99) {
            getSwiat().UsunOrganizm(this);
            Komentator.DodajKomentarz(atakujacy.OrganizmToString() + " niszczy krzak wilczej jagody");
        }
        if (atakujacy.CzyJestZwierzeciem()) {
            getSwiat().UsunOrganizm(atakujacy);
            Komentator.DodajKomentarz(atakujacy.OrganizmToString() + " ginie od wilczej jagody");
        }
        return true;
    }
}
